src/internal/observe/observe.go

```diff
@@ -117,6 +117,10 @@ func (o *observer) resetWriter(ctx context.Context) {
 	// 	opts = append(opts, mpb.WithAutoRefresh())
 	// }
 
+	if o.mp != nil {
+		logger.Ctx(ctx).Infow("resetting progress bars", "bar_count", o.mp.BarCount())
+	}
+
 	o.mp = mpb.NewWithContext(ctx, opts...)
```

Comment:
nit: should we move `o.mp = mpb.NewWithContext(ctx, opts...)` into `if o.mp != nil`  block? 

or is the intention to override `o.mp` at line#124?

Reply:
Good question: the intention is always to override.